checked_message.go

```diff
@@ -74,6 +85,44 @@ func (m *CheckedMessage) Write(fields ...Field) {
 	default:
 		m.logger.Log(m.lvl, m.msg, fields...)
 	}
+
+	if m.next != nil {
+		m.next.Write(fields...)
+	}
+}
+
+// Chain builds a chain of checked messages over several loggers. It returns an
+// OK message if either the prior message, or the new one returned by
+// logger.Check is OK.
+//
+// The returned message will first Write to any prior loggers before logging to
+// the newly passed logger.
+func (m *CheckedMessage) Chain(ms ...*CheckedMessage) *CheckedMessage {
+	for _, m2 := range ms {
+		if m2.OK() {
+			if m.OK() {
+				m.push(m2)
+			} else {
+				m = m2
+			}
+		}
+	}
+	return m
+}
+
+func (m *CheckedMessage) push(next *CheckedMessage) {
+	if m.tail != nil {
+		m.tail.next = next
+	} else if m.next != nil {
+		panic("invalid CheckedMessage linked list; did we loose our head?")
```

Comment:
Rather than panic, can we use `m.logger.DFatal` to log an error? I don't expect this to come up, but I'd really hate for a small error on our part to crash application processes in production.

Also, s/loose/lose.